File: bad-tests/parallel_construct_if_clause.c
#include "acc_testsuite.h"
#include <stdbool.h>

#ifndef T1
/*T1:parallel construct if clause,V:2.0-2.7*/
int test1(){
    int err = 0;
    bool isOffloading;

    #pragma acc parallel copy(isOffloading)
    {
        #pragma acc on_device(acc_device_not_host)
        {
            isOffloading = true;
        }
        #pragma acc on_device(acc_device_host)
        {
            isOffloading = false;
        }
    }

    // if the offloading fails, err will NOT remain 0
    if( !isOffloading){
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-finetuned/stage2/parallel_construct_if_clause.f90
!T1:parallel construct if clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0

        INTEGER :: x, y
        REAL(8),DIMENSION(10):: a, b
        
        !$acc data copy(a(1:10))
          a(:) = 0
          !$acc parallel loop if(0)
          DO x = 1, 10
            a(x) = a(x) + x
          END DO
        !$acc end data

        DO x = 1, 10
          IF (abs(a(x)) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM